### PR TITLE
Absolute and optional links that open in new tab

### LINKED ISSUE #101: Make links absolute and open in new tab

### LINKED ISSUE #102: Make link field optional

### PR DESCRIPTION
**Pull request by maxmwang**
Resolves #101, resolves #102

### CODE REVIEW

File: client/src/components/Wishlist/Wishlist.jsx [hunk 1]
```diff
@@ -168,11 +168,20 @@ class Wishlist extends Component {
       for (let item of this.props.items) {
         const styleTemp = item.style ? item.style : 'N/A';
         const notesTemp = item.notes ? item.notes : 'N/A';
+        let linkTemp;
+        if (item.link.startsWith('http://') || item.link.startsWith('https://')) {
+          linkTemp = item.link;
+        } else {
+          linkTemp = `//${item.link}`;
```
Comment (coconut750750) [inline client/src/components/Wishlist/Wishlist.jsx:175]: if the link isn't valid, can we skip adding the `<a>`?

Comment (maxmwang) [reply client/src/components/Wishlist/Wishlist.jsx:175]: We should probably warn users if the link is invalid when they initially add the item then.

**Comment by maxmwang** [comment]
Resolves #102

File: client/src/components/Wishlist/Wishlist.jsx (hunk 1 above)
Comment (coconut750750) [reply client/src/components/Wishlist/Wishlist.jsx]: what if a user doesnt have a link? we shouldnt make the link required

Comment (maxmwang) [reply client/src/components/Wishlist/Wishlist.jsx]: Added

**Comment by maxmwang** [comment]
Enabled users to edit the item's name in 3f26e07, but would be bad if users completely changed an item after it had been claimed. Perhaps revert back to not allowing users to edit item's names?

Thought it was necessary after I made a typo making a test item and realized I couldn't edit the name of the item.

File: client/src/components/Wishlist/Wishlist.jsx [hunk 2]
```diff
@@ -158,12 +172,16 @@ class Wishlist extends Component {
       for (let item of this.props.items) {
         const styleTemp = item.style ? item.style : 'N/A';
         const notesTemp = item.notes ? item.notes : 'N/A';
+        let linkTemp;
+        if (item.link.startsWith('http://') || item.link.startsWith('https://')) {
```
Comment (coconut750750) [inline client/src/components/Wishlist/Wishlist.jsx:176]: can we do something like [this](https://stackoverflow.com/questions/5717093/check-if-a-javascript-string-is-a-url) to check if the link is a valid url?

File: client/src/components/Wishlist/Wishlist.jsx [hunk 3]
```diff
@@ -268,6 +285,15 @@ class Wishlist extends Component {
           <h2>Edit Item</h2>
 
           <p>
+            <input
```
Comment (coconut750750) [inline client/src/components/Wishlist/Wishlist.jsx:288]: yeah let's not make the item's name editable

File: client/src/components/Wishlist/Wishlist.jsx [hunk 4]
```diff
@@ -50,17 +50,27 @@ class Wishlist extends Component {
...
       missingInput = 'Price';
-    } else if (!this.state.input.link) {
-      missingInput = 'Link';
     }
     if (missingInput) {
       this.setState({ errorMsg: `${missingInput} is required.` });
+      return false;
+    }
+
+    if (this.state.input.link && (!this.state.input.link.startsWith('http') || !this.state.input.link.startsWith('https'))) {
```
Comment (coconut750750) [inline client/src/components/Wishlist/Wishlist.jsx:65]: for the second part of the if statement, let's also use [this](https://stackoverflow.com/questions/5717093/check-if-a-javascript-string-is-a-url). the error message can be "Invalid Link" or something

File: client/src/components/Wishlist/Wishlist.jsx (hunk 2 above)
Comment (maxmwang) [reply client/src/components/Wishlist/Wishlist.jsx]: Added

File: client/src/components/Wishlist/Wishlist.jsx (hunk 3 above)
Comment (maxmwang) [reply client/src/components/Wishlist/Wishlist.jsx]: Added

File: client/src/components/Wishlist/Wishlist.jsx (hunk 4 above)
Comment (maxmwang) [reply client/src/components/Wishlist/Wishlist.jsx]: Added

**Review by coconut750750 — APPROVED** [review]

**Pull request merged**